mobile/ios/ios-core.ts

```diff
@@ -980,44 +984,100 @@ export class PlistService implements Mobile.IiOSDeviceSocket {
 	}
 }
 
-export class GDBServer implements Mobile.IGDBServer {
-	private socket: Mobile.IiOSDeviceSocket  = null;
+function getCharacterCodePoint(ch: string) {
+	assert.equal(ch.length, 1);
 
-	constructor(private service: number,
-		private $injector: IInjector) {
+	var code = ch.charCodeAt(0);
+	assert.ok(!(0xD800 <= code && code <= 0xffff));
 
-		if(hostInfo.isWindows()) {
-			this.socket = this.$injector.resolve(WinSocket, {service: this.service, format: 0});
-		} else if(hostInfo.isDarwin()) {
-			this.socket = this.$injector.resolve(PosixSocket, {service: this.service, format: CoreTypes.kCFPropertyListXMLFormat_v1_0});
+	return code;
+}
+
+util.inherits(GDBStandardOutputAdapter, stream.Transform);
+function GDBStandardOutputAdapter(opts?: stream.TransformOptions) {
+	stream.Transform.call(this, opts);
+	this.utf8StringDecoder = new string_decoder.StringDecoder("utf8");
+}
+GDBStandardOutputAdapter.prototype._transform = function(packet: Buffer, encoding: string, done: Function): void {
+	var result = "";
+
+	for (var i = 0; i < packet.length; i++) {
+		if (packet[i] === getCharacterCodePoint("$")) {
+			var start = ++i;
+
+			while (packet[i] !== getCharacterCodePoint("#")) {
+				i++;
+			}
+			var end = i;
+
+			// Skip checksum
+			i++;
+			i++;
+
+			if (!(packet[start] === getCharacterCodePoint("O") && packet[start + 1] !== getCharacterCodePoint("K"))) {
+				continue;
+			}
+			start++;
+
+			var hexString = packet.toString("ascii", start, end);
+			var hex = new Buffer(hexString, "hex");
+			result += this.utf8StringDecoder.write(hex);
+		}
+	}
+
+	done(null, result);
+};
+
+util.inherits(GDBSignalWatcher, stream.Writable);
+function GDBSignalWatcher(opts?: stream.WritableOptions) {
+	stream.Writable.call(this, opts);
+}
+GDBSignalWatcher.prototype._write = function(packet: Buffer, encoding: string, callback: Function) {
+	for (var i = 0; i < packet.length - 2; i++) {
+		if (packet[i] === getCharacterCodePoint("$") && (packet[i + 1] === getCharacterCodePoint("T") || packet[i + 1] === getCharacterCodePoint("S"))) {
+			// SIGKILL
+			if (packet[i + 2] === getCharacterCodePoint("9")) {
```

Comment:
If i is packet.length -2, then this 'if' will read after the buffer end. Is this possible?